tests/crypto_test.py

```diff
@@ -26,3 +33,65 @@ def test_returned_string_is_lower_case(self):
     def test_raises_for_unknown_hash_name(self):
         with self.assertRaises(GPGProblem):
             crypto.RFC3156_micalg_from_algo(gpgme.MD_NONE)
+
+
+class TestSignature(utilities.TestCaseClassCleanup):
+
+    FPR = "F74091D4133F87D56B5D343C1974EC55FBC2D660"
+
+    @classmethod
+    def setUpClass(cls):
+        # Create a temporary directory to use as gnupg's home directory. This
+        # allows us to import keys without dirtying the user's keyring
+        home = tempfile.mkdtemp()
+        cls.addClassCleanup(shutil.rmtree, home)
+        mock_home = mock.patch.dict(os.environ, {'GNUPGHOME': home})
+        mock_home.start()
+        cls.addClassCleanup(mock_home.stop)
+
+        # create a single context to use class wide.
+        cls.ctx = gpgme.Context()
+        cls.ctx.armor = True
+
+        # Add the public and private keys. They have no password
+        here = os.path.dirname(__file__)
+        with open(os.path.join(here, 'static/pub.gpg')) as f:
+            cls.ctx.import_(f)
+        with open(os.path.join(here, 'static/sec.gpg')) as f:
+            cls.ctx.import_(f)
+
+        cls.key = cls.ctx.get_key(cls.FPR)
+
+    def test_detached_signature_for(self):
+        to_sign = "this is some text.\nit is more than nothing.\n"
+        _, detached = crypto.detached_signature_for(to_sign, self.key)
+
+        with tempfile.NamedTemporaryFile(delete=False) as f:
+            f.write(detached)
+            sig = f.name
+        self.addCleanup(os.unlink, f.name)
+
+        with tempfile.NamedTemporaryFile(delete=False) as f:
+            f.write(to_sign)
+            text = f.name
+        self.addCleanup(os.unlink, f.name)
+
+        res = subprocess.check_call(['gpg', '--verify', sig, text])
```

Comment:
I guess I could just use gpgme to verify the data instead of writing it to a file and using gpg directly (or gpg after I get that work done), but if you're cool with it for now this is a valid approach. I can add a TODO.